CONTRIBUTING.md

```diff
@@ -86,37 +86,52 @@ They are written in [Cypress](https://www.cypress.io/).
 
 Visual regression tests are run for all "All Combinations" stories, or you can add specs individually for other stories.
 
-You can run UI tests as the CI would with:
+Docker (and Compose) is used for running E2E tests, so first install it via Homebrew:
+
+    brew install --cask docker
+
+Then (only once) build the local container:
+
+    docker-compose build
+
+Now you can run UI tests with:
 
     yarn e2e
 
-That runs all specs and generates coverage reports.
+This runs all specs and generates coverage reports.
+
+For writing tests locally, you will need to allow communication between Cypress within Docker container and Cypress GUI on the host system using XQuartz.
 
-Beware that environment differences between CI and your local machine might be enough for a diff to be detected.
+Install XQuartz via Homebrew:
 
-For writing tests locally:
+    brew install --cask xquartz
 
-1. You can serve the app in E2E mode with:
+Restart computer after installation.
 
-   `yarn e2e:serve`
+Start XQuartz from command line, and make sure "Allow connections from network clients" is enabled in the preferences under "Security" tab:
 
-2. Then in another terminal open Cypress with:
+    open -a XQuartz
 
-   `yarn cypress open --env failOnSnapshotDiff=false`
+Get the IP address of the host machine and allow X11 to accept incoming connections from that IP address:
 
-Alternatively you could use a single terminal with:
+    IP=$(ipconfig getifaddr en0)
+    /usr/X11/bin/xhost + $IP
 
-    yarn concurrently -k -n ,cypress:open yarn:e2e:serve "yarn cypress open --env failOnSnapshotDiff=false"
+Then open Cypress GUI (using same terminal window that had the IP address set):
 
-For a nicer experience, [you can report snapshot diffs in your test results to console](https://github.com/jaredpalmer/cypress-image-snapshot#reporter):
+    DISPLAY=$IP:0 CYPRESS_baseUrl=http://host.docker.internal:6006 docker-compose -f docker-compose.yml -f docker-compose.open.yml up --abort-on-container-exit --exit-code-from cypress
```

Comment:
Currently I placed this in docs (full command).

Happy to place within `package.json` scripts, if you think it's better place for that?